src/shared/selinux-util.c

```diff
@@ -562,6 +562,21 @@ int mac_selinux_create_file_prepare(const char *path, mode_t mode) {
 #endif
 }
 
+int mac_selinux_create_file_prepare_label(const char *path, const char *label) {
```

Comment:
@doverride suggested to not fail on invalid lables and @pebenito also [commented](https://github.com/systemd/systemd/pull/20767#issuecomment-921885329)

>>What if the image doesn't have labels or labels are invalid?

> In this case, the code would have to fall back to the existing behavior.


Maybe handle the case where [`security_check_context_raw(label) != 0`](https://manpages.debian.org/experimental/libselinux1-dev/security_check_context_raw.3.en.html)?

Reply:
The coding style is that it's the caller that decides how to handle failures, and as requested the caller logs and moves on